main/src/main/scala/sbt/Main.scala

```diff
@@ -92,7 +92,7 @@ final class xMain extends xsbti.AppMain {
   private def endsWithBoot(state: State) = state.remainingCommands.lastOption exists (_.commandLine == BootCommand)
 
   private def notifyUsersAboutShell(state: State) =
-    if (isInteractive && !hasCommand(state, Shell) && !hasCommand(state, Server) && !endsWithBoot(state)) {
+    if (isInteractive && !hasCommand(state, Shell) && !hasCommand(state, OldShell) && !endsWithBoot(state)) {
```

Comment:
Ah it also has Shell. @eed3si9n if possible could you draw attention to these details when forward-porting?